common/src/main/java/org/opensearch/ml/common/transport/connector/MLConnectorDeleteRequest.java

```diff
@@ -22,24 +24,41 @@
 import lombok.Builder;
 import lombok.Getter;
 
+@Getter
 public class MLConnectorDeleteRequest extends ActionRequest {
-    @Getter
-    String connectorId;
+    private final String connectorId;
+    private final String tenantId;
 
     @Builder
+    public MLConnectorDeleteRequest(String connectorId, String tenantId) {
+        this.connectorId = connectorId;
+        this.tenantId = tenantId;
+    }
+
     public MLConnectorDeleteRequest(String connectorId) {
         this.connectorId = connectorId;
+        this.tenantId = null;
     }
 
     public MLConnectorDeleteRequest(StreamInput input) throws IOException {
         super(input);
+        Version streamInputVersion = input.getVersion();
         this.connectorId = input.readString();
+        if (streamInputVersion.onOrAfter(VERSION_2_19_0)) {
+            this.tenantId = input.readOptionalString();
+        } else {
+            this.tenantId = null;
+        }
```

Comment:
If you choose to keep `tenantId` final this may be more readable with a ternary operator